diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt

```diff
@@ -459,19 +630,26 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
     }
 
     /**
-     * Finds the first binary expression closer to the separator
+     * Finds the first binary expression or dat closer to the separator
      */
     @Suppress("UnsafeCallOnNullableType")
-    private fun searchRightSplitInBinaryExpression(parent: ASTNode, configuration: LineLengthConfiguration): Pair<ASTNode, Int>? {
-        val binList: MutableList<ASTNode> = mutableListOf()
-        searchBinaryExpression(parent, binList)
-        return binList.map {
-            it to positionByOffset(it.getFirstChildWithType(OPERATION_REFERENCE)!!.startOffset).second
+    private fun searchRightSplitAfterType(
+        parent: ASTNode,
+        configuration: LineLengthConfiguration,
+        type: IElementType
+    ): Pair<ASTNode, Int>? {
+        val list: MutableList<ASTNode> = mutableListOf()
+        when (type) {
+            OPERATION_REFERENCE -> searchBinaryExpression(parent, list)
+            DOT, SAFE_ACCESS -> searchDotOrSafeAccess(parent, list)
+        }
+        return list.map {
```

Comment:
I tend to avoid direct `list.map()` calls w/o creating a sequence first: otherwise, each subsequent call produces an intermediate container (`List<T>` -> `List<U>` -> `List<V>` -> ...). Sequences, on the other hand, are _lazy_ and consume next to no extra memory. Better:

```kotlin
return list
.asSequence()
.map { /* ... */ }
.sortedBy { /* ... */ }
.toList() // You can't reverse a sequence, so here you do have to convert it to list first
.reversed()
.firstOrNull { /* ... */ }
```

Even better (w/o intermediate lists at all):

```kotlin
return list
.asSequence()
.map { /* ... */ }
.sortedBy { /* ... */ }
.lastOrNull { /* ... */ }
```